### PR TITLE
fix gazebo_ros_power_monitor plugin

### PR DESCRIPTION
**Pull request by furushchev**
This fixes a crash on 32 bit range error in `pr2_gazebo`.

### CODE REVIEW

File: pr2_gazebo_plugins/src/gazebo_ros_power_monitor.cpp [hunk 1]
```diff
@@ -42,106 +42,167 @@ using namespace std;
...
 
 GazeboRosPowerMonitor::GazeboRosPowerMonitor()
-{
-    // model_ = dynamic_cast<Model*>(parent);
-    // if (!model_)
-    //     gzthrow("GazeboRosPowerMonitor controller requires a Model as its parent");
-
-    // // Initialize parameters
-    // Param::Begin(&parameters);
-    // robot_namespace_param_   = new ParamT<string>("robotNamespace",     "/",           0);
```
Comment (furushchev) [inline pr2_gazebo_plugins/src/gazebo_ros_power_monitor.cpp:52]: Here any parameters should have been read but no parameters actually read from urdf and all the parameters are zero which can cause zero division error and therefore this plugin publishes wrong trivial data.

**Comment by furushchev** [comment]
kindly ping to maintainers. Without this patch, spawning PR2 will cause a crash on gazebo.

**Review by v4hn — APPROVED** [review]
Sorry @furushchev , I postponed reviewing this because it's rather big changes to fix something seemingly easy.

You rearranged a number of lines for no obvious reason, deleted some others and you did not split the changes into multiple commits that explain why you changed things. Please try to do this so we can review more easily!
"fix a crash" is no good description of rewriting half the plugin (which was fully justified now that I looked at it).

I approve

**Comment by furushchev** [comment]
@v4hn Sorry for late response and thank you very much for your review!
I'll make more detailed description from the next time! 👍 

@k-okada Please could you merge this? (I'm not sure I should mention to you though)

**Comment by k-okada** [comment]
@furushchev please revert unnecessary changes.

**Comment by v4hn** [comment]
> @furushchev please revert unnecessary changes.

To me the "unnecessary" changes are quite an improvement and I would be happy to keep them.

However, if you think so too, it would be good to have them in separate commits.

**Comment by furushchev** [comment]
@k-okada @v4hn I forgot to inform you, I cleaned up this pull request.

**Comment by furushchev** [comment]
@k-okada ping

**Pull request merged**